File: alarm/src/main/java/com/pape/alarm/AsyncHandler.java
package com.pape.alarm;

import android.os.Handler;
import android.os.HandlerThread;

public class AsyncHandler {
	
	private static final HandlerThread sHandlerThread=new HandlerThread("AsyncHandler");
	private static final Handler sHandler;
	
	static {
		sHandlerThread.start();
		sHandler=new Handler(sHandlerThread.getLooper());
	}
	
	public static void post(Runnable r){
		sHandler.post(r);
	}
	
	private AsyncHandler(){}
}

File: alarm/src/main/java/com/pape/alarm/AsyncHandler.kt
package com.pape.alarm

import android.os.Handler
import android.os.HandlerThread

object AsyncHandler {

    private val sHandlerThread = HandlerThread("AsyncHandler")
    private val sHandler: Handler

    init {
        sHandlerThread.start()
        sHandler = Handler(sHandlerThread.looper)
    }

    fun post(r: Runnable) {
        sHandler.post(r)
    }
}
